matrix_content_scanner/logutils.py

```diff
@@ -0,0 +1,65 @@
+#  Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+from contextvars import ContextVar
+from typing import Any, Callable
+
+from twisted.web.http import Request
+
+# The serverName/mediaId path of the media.
+media_path: ContextVar[str] = ContextVar("media_path")
+# The request being performed (download, thumbnail, scan, etc).
+request_type: ContextVar[str] = ContextVar("request_type")
+
+
+def get_factory() -> Callable[..., logging.LogRecord]:
```

Comment:
docstring please

Reply:
`get_factory` sounds like it doesn't have any side-effects, and is essentially idempotent. However, under your implementation, it will create a new factory (chained to the previous one).

Accordingly, `make_factory` or something might be a better name.

Alternatively, make it cache the result:

```suggestion
_factory = None

def get_factory() -> Callable[..., logging.LogRecord]:
    global _factory
    if _factory != None:
        return _factory

```

etc. Oh, and if you're targetting python 3.9 and later, there's functools.cache to do this for you.